template/node/src/service.rs

```diff
@@ -201,12 +202,15 @@ async fn build_relay_chain_interface(
 ///
 /// This is the actual implementation that is abstract over the executor and the runtime api.
 #[sc_tracing::logging::prefix_logs_with("Parachain")]
-async fn start_node_impl(
+async fn start_node_impl<N>(
```

Comment:
Why does this need a generic? Isn't `N` always `sc_network::NetworkWorker`?

Reply:
not always: https://github.com/paritytech/polkadot-sdk/blob/313fe0f9a277f27a4228634f0fb15a1c3fa21271/substrate/bin/node/cli/src/service.rs#L813

Reply:
I did not implement the litep2p network one, but I guess better leave it prepared for it

Reply:
Ok sounds good